service/controller/app/resource_set.go

```diff
@@ -283,11 +284,18 @@ func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
 			return false
 		}
 
-		if key.VersionLabel(cr) == project.AppVersion() {
-			return true
+		// When app-operator is deployed as a unique app it only processes
+		// control plane app CRs. These CRs always have the version label
+		// app-operator.giantswarm.io/verson: 0.0.0
+		if config.UniqueApp {
+			return key.VersionLabel(cr) == project.AppControlPlaneVersion()
 		}
 
-		return false
+		// Currently tenant cluster app CRs always have the version label
+		// app-operator.giantswarm.io/verson: 1.0.0
+		// This hardcoding will be removed in a future release and we will then
+		// use project.Version().
+		return key.VersionLabel(cr) == project.AppTenantVersion()
```

Comment:
Once we remove the hardcoding we can use `project.Version()` here. 

@kopiczko I know the handlerFunc is being removed. But we'll use the replacement once we upgrade to the latest operatorkit.